lib/file.js

```diff
@@ -147,6 +150,62 @@ class File extends EventEmitter {
     render.render(this, elem, opts, cb)
   }
 
+  _serve (req) {
```

Comment:
I actually think all of this logic belongs with the onMessage impl, not in file.

Reply:
I think this is the right place for it, but it doesnt seem right either?... it's really up to the maintainers to decide on this